src/cirrus/lib/process_payload.py

```diff
@@ -198,55 +219,90 @@ def __call__(self, wfem) -> str | None:
         if not payload_bucket:
             raise ValueError("env var CIRRUS_PAYLOAD_BUCKET must be defined")
 
-        arn = os.getenv("CIRRUS_BASE_WORKFLOW_ARN") + self.process["workflow"]
+        state_machine_arn, _, execution_name = execution_arn.rpartition(":")
         url = f"s3://{payload_bucket}/{self['id']}/input.json"
 
         # start workflow
-        try:
-            # add input payload to s3
-            s3().upload_json(self, url)
 
-            # create DynamoDB record
-            # -> this overwrites existing states other than PROCESSING
-            wfem.claim_processing(self["id"], payload_url=url)
-
-            # invoke step function
-            self.logger.debug("Running Step Function %s", arn)
-            exe_response = get_client("stepfunctions").start_execution(
-                stateMachineArn=arn,
-                input=json.dumps(self.get_payload()),
-            )
+        # add input payload to s3
+        s3().upload_json(self, url)
 
-            # add execution to DynamoDB record
-            wfem.started_processing(
+        # create DynamoDB record
+        try:
+            # -> overwrites states other than PROCESSING and CLAIMED
+            wfem.claim_processing(
                 self["id"],
-                exe_response["executionArn"],
                 payload_url=url,
+                execution_arn=execution_arn,
             )
-
-            return self["id"]
         except ClientError as e:
             if e.response["Error"]["Code"] == "ConditionalCheckFailedException":
-                wfem.skipping(self["id"], state=StateEnum.PROCESSING, payload_url=url)
+                # as previous_state was not CLAIMED, this should be PROCESSING,
+                # hence skipping w/o need for other action.  Double checking via
+                # exception response.
+                db_state = (
+                    StateEnum(
+                        e.response["Item"]["state_updated"]["S"].split("_")[0],
+                    )
+                    if "Item" in e.response
+                    else StateEnum.PROCESSING
+                )
+
+                wfem.skipping(
+                    self["id"],
+                    state=db_state,
+                    payload_url=url,
+                    message=f"state before claim attempt was {previous_state}",
+                )
                 return None
```

Comment:
this needs differential behaviors based on the state returned in the exception:

1. if claimed take the execution from the returned db record
2. if processing skip (renamed to `already_in`), return None